core/src/main/java/com/graphhopper/routing/lm/LMRoutingAlgorithmFactory.java

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.graphhopper.routing.lm;
+
+import com.graphhopper.routing.*;
+import com.graphhopper.routing.weighting.Weighting;
+import com.graphhopper.storage.Graph;
+import com.graphhopper.util.Parameters;
+
+public class LMRoutingAlgorithmFactory implements RoutingAlgorithmFactory {
+    private final RoutingAlgorithmFactory defaultAlgoFactory;
+    private final LandmarkStorage lms;
+    private final Weighting prepareWeighting;
+    private final int defaultActiveLandmarks;
+    private final int numBaseNodes;
+
+    public LMRoutingAlgorithmFactory(LandmarkStorage lms, int defaultActiveLandmarks, int numBaseNodes) {
```

Comment:
The minimal requirement to create an LM routing factory should be a `LandmarkStorage`, similar as a CH routing factory needs a `CHGraph` (the pre-processed data). `defaultActiveLandmarks` is also needed to provide the default number of active landmarks which is used when `AlgoOpts` does not specify a value for this in `createAlgo()`. The `numBaseNodes` paramater is quite ugly however, maybe we can somehow store/retrieve this information from `LandmarkStorage` as well?

Reply:
And we should avoid successive parameters of the same type if possible.

Reply:
Ok, if the `numBaseNodes` parameter stays (which I do not hope) we can at least change the order of parameters

Reply:
> The numBaseNodes paramater is quite ugly however, maybe we can somehow store/retrieve this information from LandmarkStorage as well?

Yes, this is easy as the GraphHopperStorage is available in LandmarkStorage

Reply:
Oh right I must have missed this somehow -> removed the parameter.